#!/usr/bin/env python3
PACKAGE = "aruco_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

corner_enum = gen.enum([    gen.const("CORNER_SUBPIX", int_t, 0, "subpix corner refinement"),
                            gen.const("CORNER_LINES", int_t, 1, "lines corner refinement"),
                            gen.const("CORNER_NONE", int_t, 2, "No corner refinement")  ],
                        "enum to set aruco corner refinement method")

gen.add("corner_refinement_method", int_t, 0, "corner refinement technique", 0, 0, 2, edit_method=corner_enum)

mode_enum = gen.enum([  gen.const("DM_NORMAL", int_t, 0, "normal mode"),
                        gen.const("DM_FAST", int_t, 1, "fast mode"),
                        gen.const("DM_VIDEO_FAST", int_t, 2, "faster mode for videos")  ],
                      "Detection mode of aruco markers")

gen.add("detection_mode", int_t, 0, "mode of detection", 0, 0, 2, edit_method=mode_enum)

gen.add("min_marker_size", double_t, 0, "minimum marker size (proportion)", 0, 0, 1)

exit(gen.generate(PACKAGE, "aruco_ros", "ArucoThreshold"))
